src/routes/goals/handlers.js

```diff
@@ -30,3 +59,36 @@ export async function changeGoalStatus(req, res) {
     await handleErrors(req, res, error, logContext);
   }
 }
+
+export async function deleteGoal(req, res) {
+  try {
+    const { goalId } = req.params;
+    const { regionId } = req.body;
+
+    const user = await userById(req.session.userId);
+
+    const policy = new Goal(user, {
```

Comment:
I would expect this to be passed an instance of a goal. Also would allow us to not care about regionId in the body at all, which cleans up the call (you only need the goal id to delete). Plus someone could call the API with a goal id from a different region and be able to delete the goal

Reply:
Do you think I should put the sequelize logic for looking up the goal region in this file?

Reply:
right right, but goals don't have regions, so I'd need to get the grants attached, loop through and check that at least one matches the user's permissions. But it sounds like maybe the services is the best place for that, not the policy

Reply:
Oh gotcha. Yeah I'd probably add something to the goals service going from the goal to all grants to grab all regions. This will change in the future to be a bit nicer once a goal has a single grant